### DISCUSSION TITLE
使用zig跨平台编译lua

### DISCUSSION THREAD

**Discussion by hanxi** [zigcc/forum#141] [作品分享]
项目地址: https://github.com/hanxi/zig-build-lua

zig 支持了包管理，参考这个项目 https://github.com/brickBrackBro/lua 把工程里的 lua 源码清理了。zig 作为 makefile 的替代品真的很棒，关键是一次能编译多个平台的产物，这点很像 go, 简化了 c 跨平台编译的流程。

今天也学到了使用 rdynamic 来实现 -Wl,-E 的效果，也就是在可执行文件导出函数接口，直接加 -Wl,-E 是不生效的。

**Comment by jiacai2050** [comment]
> 今天也学到了使用 rdynamic 来实现 -Wl,-E 的效果，也就是在可执行文件导出函数接口，直接加 -Wl,-E 是不生效的。

这个能展开介绍一下嘛？之前没接触过这个。

**Reply by hanxi** [reply]
lua在非windows平台默认是只生成一个可执行文件的，没有so文件的。然后lua又是支持加载动态库so文件的，动态库的实现都会调用lua里接口函数，因此需要在可执行文件里导出lua的接口给动态库使用。在lua的makefile里是通过 -Wl,-E 做到的，在 zig 里需要使用 rdynamic 来实现一样的效果。具体见 https://github.com/ziglang/zig/issues/6804#issuecomment-2293315392
